core/test/utils/matrix_generator.hpp

```diff
@@ -128,6 +126,65 @@ std::unique_ptr<MatrixType> generate_random_matrix(
 }
 
 
+/**
+ * Generates a random lower triangular matrix.
+ *
+ * @tparam MatrixType  type of matrix to generate (matrix::Dense must implement
+ *                     the interface `ConvertibleTo<MatrixType>`)
+ * @tparam NonzeroDistribution  type of nonzero distribution
+ * @tparam ValueDistribution  type of value distribution
+ * @tparam Engine  type of random engine
+ *
+ * @param num_rows  number of rows
+ * @param num_cols  number of colums
+ * @param nonzero_dist  distribution of nonzeros per row
+ * @param value_dist  distribution of matrix values
+ * @param engine  a random engine
+ * @param exec  executor where the matrix should be allocated
+ * @param args  additional arguments for the matrix constructor
+ */
+template <typename MatrixType = matrix::Dense<>, typename NonzeroDistribution,
+          typename ValueDistribution, typename Engine, typename... MatrixArgs>
+std::unique_ptr<MatrixType> generate_random_lower_triangular_matrix(
```

Comment:
Since you said that some triangular matrices have only `1`s on their diagonal, while others don't, I would appreciate a way of generating both versions (with only `1`s on the diagonal, while the other has random numbers there) with this single function.